### PR TITLE
RD-1770 RD-1772 Added support for saving modified filter

### PR DESCRIPTION
**Pull request by kubama**
Please note the button added in the bottom left corner.
image

System tests: https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/829/

### CODE REVIEW

**Comment by kubama** [comment]
System tests covering most recent commit: https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/830/

**Review by stepek — APPROVED** [review]
Code LGTM!

**Comment by Gelio** [comment]
I'll review the PR later today as I have quite a few meetings early :slightly_frowning_face:

File: widgets/common/src/deploymentsView/header/FilterModal.tsx [hunk 1]
```diff
@@ -110,12 +114,31 @@ const FilterModal: FunctionComponent<FilterModalProps> = ({
...
+            return;
+        }
+
+        setFilterSaving();
+        new FilterActions(toolbox)
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            .doUpdate(filterId.value!, filterRules.value)
+            .then(() => filterDirty.set(false))
+            .catch(setMessageAsError)
+            .finally(unsetFilterSaving);
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/header/FilterModal.tsx:135]: Minor/general:

I'm wondering what should happen if the user closes the modal when the request is in progress :thinking:

These can cause a problem with _Cannot update React state after a component is unmounted_

File: widgets/common/src/deploymentsView/header/FilterModal.tsx [hunk 2]
```diff
@@ -154,6 +177,12 @@ const FilterModal: FunctionComponent<FilterModalProps> = ({
             </Modal.Content>
 
             <Modal.Actions>
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/header/FilterModal.tsx:179]: I believe the buttons should be disabled when the "Save" request is in flight. At the moment, the user can click "Cancel" during saving, which IMO results in undefined behavior (broken UI :smile: )

Unfortunately, the `LoadingOverlay` does not cover the actions at the bottom

**Review by Gelio — COMMENTED** [review]
Looks good to me overall :+1: I've also tested it and seen it work - works well :smile: 

I just added 2 points about UX

File: widgets/common/src/deploymentsView/header/FilterModal.tsx (hunk 2 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/FilterModal.tsx]: Good point, I extended the overlay over action buttons.

File: widgets/common/src/deploymentsView/header/FilterModal.tsx (hunk 1 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/FilterModal.tsx:144]: I believe with https://github.com/cloudify-cosmo/cloudify-stage/pull/1453#discussion_r657822786 addressed there should be no longer such risk 🙂.

Comment (Gelio) [reply widgets/common/src/deploymentsView/header/FilterModal.tsx:144]: Yup, sounds good :+1:

**Review by Gelio — APPROVED** [review]
Looks great :+1: :ok_hand:

**Pull request merged**